### ISSUE TITLE
Add --oauth-service-account-email and --oauth-token-scope to cloud scheduler resource

### ISSUE THREAD

**Issue by spacetj** [terraform-providers/terraform-provider-google#4089]
### Description

Add --oauth-service-account-email and --oauth-token-scope to cloud scheduler resource so that we can use cloud scheduler to make gcp rest calls.

### New or Affected Resource(s)

google_cloud_scheduler_job

### Potential Terraform Configuration

```tf
resource "google_cloud_scheduler_job" "job" {
  name     = "test-job"
  schedule = "*/4 * * * *"
  oauth-service-account-email = "gcloud-sa@gcloud.com"
  oauth-token-scope = "https://www.googleapis.com/auth/cloud-platform"

  http_target {
    uri = "https://region-gcp_project.cloudfunctions.net/func_name"
    http_method = "POST"
   }
}
```

---
# Revised implementation Taken from @francoissera comment below.
```HCL
resource "google_cloud_scheduler_job" "job" {
  name     = "test-job"
  schedule = "*/4 * * * *"

  http_target {
    uri = "https://region-gcp_project.cloudfunctions.net/func_name"
    http_method = "POST"
    oauthToken = {
        serviceAccountEmail = "myservice_account_email}"
        audience = "https://region-gcp_project.cloudfunctions.net/func_name"
     }
   }
 }
```

**Comment by francoisserra** [comment]
I think `oauthToken` and `oidcToken` are part of HttpTarget and not AppEngineHttpTarget, as stated in the API doc https://cloud.google.com/scheduler/docs/reference/rest/v1/projects.locations.jobs#httptarget

for example, to schedule the call of a http triggered cloud function with authentication enabled:

```
resource "google_cloud_scheduler_job" "job" {
  name     = "test-job"
  schedule = "*/4 * * * *"

  http_target {
    uri = "https://region-gcp_project.cloudfunctions.net/func_name"
    http_method = "POST"
    oidcToken = {
                serviceAccountEmail = "myservice_account_email}"
                audience = "https://region-gcp_project.cloudfunctions.net/func_name"
     }
   }
 }
```

**Comment by spacetj** [comment]
You're right @francoisserra. I have updated and credited my initial issue to take in your changes. Thanks for the feedback.

**Comment by chriswacker** [comment]
I would really like to see this so Cloud Scheduler jobs that call authenticated Cloud Run apps can be fully managed with Terraform.

**Comment by SP3269** [comment]
Should that be "oauth-token-**scopes**", plural and array, and not a singular "oauth-token-scope"? E.g.
```
oauth-token-scopes = [ "https://www.googleapis.com/auth/admin.directory.group", "https://www.googleapis.com/auth/admin.directory.user", ]
```
More often than not I use narrow scopes for my applications, have to use multiple.

**Comment by megan07** [comment]
Thanks @spacetj for opening this issue. It's been closed with #4222 and will be released shortly.
